geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,47 +238,89 @@ object IngestCommand extends LazyLogging {
+            case _: Exception => throw new ParameterException("Unable to read from stdin")
+          }
+        }
+
+        open = openStdin
       }
-      val opened = ListBuffer.empty[CloseableIterator[InputStream]]
-      def open(): InputStream = {
-        val streams = file.open.map(_._2)
-        opened += streams
-        if (streams.hasNext) { streams.next } else {
+      // try to infer the schema from the input files themselves
+      else {
+        Command.user.info("No converter defined - will attempt to detect schema from input files")
+        file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
           throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
         }
+        val opened = ListBuffer.empty[CloseableIterator[InputStream]]
+
+        def openFile(): InputStream = {
+          val streams = file.open.map(_._2)
+          opened += streams
+          if (streams.hasNext) {
+            streams.next
+          } else {
+            throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+          }
+        }
+
+        open = openFile
       }
+
       val (inferredSft, inferredConverter) = try {
         val opt = format match {
-          case None      => SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
-          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          case None => SimpleFeatureConverter.infer(open, Option(sft))
+          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), None)
         }
         opt.getOrElse {
           throw new ParameterException("Could not determine converter from inputs - please specify a converter")
         }
       } finally {
-        CloseWithLogging(opened)
+        CloseWithLogging(open())
       }
+
       val renderOptions = ConfigRenderOptions.concise().setFormatted(true)
       var inferredSftString: Option[String] = None
 
       if (sft == null) {
-        val typeName = Option(params.featureName).getOrElse {
-          val existing = ds.toSet[DataStore].flatMap(_.getTypeNames)
-          val fileName = Option(FilenameUtils.getBaseName(file.path))
-          val base = fileName.map(_.trim.replaceAll("[^A-Za-z0-9]+", "_")).filterNot(_.isEmpty).getOrElse("geomesa")
-          var name = base
-          var i = 0
-          while (existing.contains(name)) {
-            name = s"${base}_$i"
-            i += 1
+
+        val typeName = if (inputs == Inputs.StdInInputs) {
+          Option(params.featureName).getOrElse("default")
+        } else {
+          Option(params.featureName).getOrElse {
+            val existing = ds.toSet[DataStore].flatMap(_.getTypeNames)
+            val fileName = Option(FilenameUtils.getBaseName(file.path))
+            val base = fileName.map(_.trim.replaceAll("[^A-Za-z0-9]+", "_")).filterNot(_.isEmpty).getOrElse("geomesa")
```

Comment:
we should probably get rid of the `getOrElse("geomesa")` here and require that they specify a type name, for the same reasons mentioned above.